ui/component/livestreamChatLayout/view.jsx

```diff
@@ -0,0 +1,375 @@
+  doCommentSocketConnect: (string) => void,
+  doCommentSocketDisconnect: (string) => void,
+  doResolveUris: (Array<string>) => void,
+  doSuperChatList: () => void,
+};
+
+export default function LivestreamChatLayout(props: Props) {
+  const {
+    claim,
+    comments: commentsByChronologicalOrder,
+    embed,
+    isPopoutWindow,
+    pinnedComments,
+    superChats: superChatsByAmount,
+    uri,
+    doCommentList,
+    doCommentSocketConnect,
+    doCommentSocketDisconnect,
+    doResolveUris,
+    doSuperChatList,
+  } = props;
+
+  const {
+    location: { pathname },
+  } = useHistory();
+
+  const isMobile = useIsMobile();
+
+  const discussionElement = document.querySelector('.livestream__comments');
+
+  const restoreScrollPos = React.useCallback(() => {
+    if (discussionElement) discussionElement.scrollTop = 0;
+  }, [discussionElement]);
+
+  const commentsRef = React.createRef();
+
+  const [viewMode, setViewMode] = React.useState(VIEW_MODES.CHAT);
+  const [scrollPos, setScrollPos] = React.useState(0);
+  const [showPinned, setShowPinned] = React.useState(true);
+  const [resolvingSuperChats, setResolvingSuperChats] = React.useState(false);
+  const [mention, setMention] = React.useState();
+  const [openedPopoutWindow, setPopoutWindow] = React.useState(false);
+  const [chatHidden, setChatHidden] = React.useState(false);
+
+  const quickMention =
+    mention && formatLbryUrlForWeb(mention).substring(1, formatLbryUrlForWeb(mention).indexOf(':') + 3);
+  const claimId = claim && claim.claim_id;
+  const commentsToDisplay = viewMode === VIEW_MODES.CHAT ? commentsByChronologicalOrder : superChatsByAmount;
+  const commentsLength = commentsToDisplay && commentsToDisplay.length;
+  const pinnedComment = pinnedComments.length > 0 ? pinnedComments[0] : null;
+
+  let superChatsChannelUrls = [];
+  let superChatsFiatAmount = 0;
+  let superChatsLBCAmount = 0;
+  if (superChatsByAmount) {
+    superChatsByAmount.forEach((superChat) => {
+      const { is_fiat: isFiat, support_amount: tipAmount, channel_url: uri } = superChat;
+
+      if (isFiat) {
+        superChatsFiatAmount = superChatsFiatAmount + tipAmount;
+      } else {
+        superChatsLBCAmount = superChatsLBCAmount + tipAmount;
+      }
+      superChatsChannelUrls.push(uri || '0');
+    });
+  }
+
+  function toggleSuperChat() {
+    if (superChatsChannelUrls && superChatsChannelUrls.length > 0) {
+      doResolveUris(superChatsChannelUrls);
+
+      if (superChatsByAmount.length > LARGE_SUPER_CHAT_LIST_THRESHOLD) {
+        setResolvingSuperChats(true);
+      }
+    }
+    setViewMode(VIEW_MODES.SUPERCHAT);
+  }
+
+  function handlePopout() {
+    const newWindow = window.open('/$/popout' + pathname, 'Popout Chat', 'height=700,width=400');
+
+    // Add function to newWindow when closed manually
+    // show the chat again and re-connect to websocket
+    newWindow.onbeforeunload = () => {
+      if (claimId) doCommentSocketConnect(claimId);
```

Comment:
Steps:
- Open popup
- Click "Close Popup" from the Main window.  You'll see the "currently waiting" viewer count increment.
- Repeat the above.  It increments forever until a refresh cleans it up.

![image](https://user-images.githubusercontent.com/64950861/149430993-8fc67a3e-8856-40c5-bcc9-3b5868095641.png)